x-pack/plugins/monitoring/server/alerts/cpu_usage_alert.ts

```diff
@@ -397,79 +333,49 @@ export class CpuUsageAlert extends BaseAlert {
     }
   }
 
-  protected processData(
-    data: AlertData[],
-    clusters: AlertCluster[],
-    services: AlertServices,
-    logger: Logger
-  ) {
+  protected processData(data: AlertData[], clusters: AlertCluster[], services: AlertServices) {
+    const currentUTC = +new Date();
     for (const cluster of clusters) {
-      const nodes = data.filter((_item) => _item.clusterUuid === cluster.clusterUuid);
-      if (nodes.length === 0) {
+      const nodes = data.filter((node) => node.clusterUuid === cluster.clusterUuid);
+      if (!nodes.length) {
         continue;
       }
 
-      const firingNodeUuids = nodes.reduce((list: string[], node) => {
-        const stat = node.meta as AlertCpuUsageNodeStats;
-        if (node.shouldFire) {
-          list.push(stat.nodeId);
-        }
-        return list;
-      }, [] as string[]);
-      firingNodeUuids.sort(); // It doesn't matter how we sort, but keep the order consistent
-      const instanceId = `${this.type}:${cluster.clusterUuid}:${firingNodeUuids.join(',')}`;
+      const instanceId = `.monitoring:${this.type}:${cluster.clusterUuid}`;
```

Comment:
While this does help with maintaining instance state for after resolving an alert, it introduces another problem. The unique instance id is where the throttling is enforced - meaning if you create an instance with a previously used id, the actions are subject to the throttle period started by the first time you used that instance.

In this case, since the instance id is based off the cluster id, this will never change from cluster to cluster, even if the number of nodes that are firing changes. 

Imagine a 3 node cluster (A, B, C) and node A is firing an alert, this instance id will be `.monitoring:cpu_usage:clusterUuid` and the actions will fire and then the throttling will start for all `.monitoring:cpu_usage:clusterUuid` instances. Now, imagine node A resolves itself, but node B starts firing an alert. This will run and try and fire the actions, but it will be subject to the throttle period (which by default is `1d`) so they wouldn't see any messaging about it.

This is why I originally did it by `firingNodeUuids` to ensure we generated a unique instance id based on what was actually firing.

I'm not sure we can go in this direction because I worry that our alerting will miss valid cases where it should send actions and we will lose trust with our users.

WDYT?